ggml-cuda.cu

```diff
@@ -4203,14 +4236,24 @@ static void silu_f32_cuda(const float * x, float * dst, const int k, cudaStream_
 
 static void norm_f32_cuda(const float * x, float * dst, const int ncols, const int nrows, cudaStream_t stream) {
     GGML_ASSERT(ncols % WARP_SIZE == 0);
-    const dim3 block_dims(WARP_SIZE, 1, 1);
-    norm_f32<<<nrows, block_dims, 0, stream>>>(x, dst, ncols);
+    if (ncols < 1024) {
```

Comment:
I think the code will not produce correct results for tensor sizes not divisible by 1024. So I think the check should instead be:

```suggestion
    if (ncols % 1024 != 0) {
```

Reply:
It produces correct results for any hidden size. I tested it with hidden size 1234 and it produces the same results as PyTorch. I think we should even remove this check `GGML_ASSERT(ncols % WARP_SIZE == 0);`.

Reply:
Yes, I think you're right. The loops only go up to `ncols` so this should make out-of-bounds memory accesses impossible.